compiler/rustc_target/src/spec/base/apple/mod.rs

```diff
@@ -97,53 +98,95 @@ impl TargetAbi {
 }
 
 fn pre_link_args(os: &'static str, arch: Arch, abi: TargetAbi) -> LinkArgs {
+    let mut args = LinkArgs::new();
```

Comment:
I think `TargetOptions::link_args` can still be used here instead of `LinkArgs::new`.